### DISCUSSION TITLE
[✨] Opt-out streaming

### DISCUSSION THREAD

**Discussion by GrandSchtroumpf** [QwikDev/qwik-evolution#133] [Proposals For Qwik]
### Is your feature request related to a problem?

A lot of issues in Qwik seems to be related to streaming because **we cannot update items that have been already rendered** and sent to the client.
I think streaming is a good feature, but it comes at a high price. And sometime we don't need streaming at all, like with SSG.

Streaming is currently preventing us to update context at render time, which has a huge impact on how we built the app: 
- limiting head update to `routeLoader$` (which also come with side effects) QwikDev/qwik#4712
- designing components in a specific matter QwikDev/qwik#4383
- cannot lazy query some data based on amount of listeners https://github.com/BuilderIO/qwik/issues/2715#issuecomment-1751990745

### Describe the solution you'd like

The function `renderToString` should not use `renderToStream` under-the-hood.
Renderer should wait for DOM to be stable with a timeout to avoid infinity loop a little bit and angular universal is working.

### Describe alternatives you've considered

Not updating the context at render time

**Comment by wmertens** [comment]
I think this can't be done easily, because the SSR code assumes that the components will not be rerendered. I might be wrong.

If it was easy to support it would make an interesting feature, although I think all the examples you give are better solved in other ways.

**Comment by wmertens** [comment]
@GrandSchtroumpf I'm looking at the SSR code and what you want would require a complete rewrite, so it would be extremely unlikely.

**Comment by wmertens** [comment]
I just realized that opt-out streaming is possible by running regular Qwik on JSDom on the server, and then pausing the container. That would be possible as a standalone package.

**Comment by GrandSchtroumpf** [comment]
That would be amazing ! Qwik would work the same in web & server (modulo differences between DOM & JSDom).
Do you have any idea how to achieve it ? I assume it involve more than just : 
```jsx
const { window } = new JSDOM('');
render(window.document, <Root />, opts)
```

**Comment by wmertens** [comment]
Seems to me that you have to await the result and then clean up the subscriptions, and then you can call root.pause() and read the outerHtml to get the container.

https://qwik.builder.io/api/qwik/#render

It wouldn't be efficient but for generating SSG that's ok

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
